### PR TITLE
Update Readme to V0.14.2

### PR DESCRIPTION
**Pull request by ArangoGutierrez**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,4 +39,4 @@ $ kubectl get no -o json | jq '.items[].metadata.labels'
 
 ```
 
-[documentation]: https://kubernetes-sigs.github.io/node-feature-discovery
+[documentation]: https://kubernetes-sigs.github.io/node-feature-discovery/v0.14
```
Comment (marquiz) [inline README.md:42]: We don't need the version here. The link will always get redirected to the latest stable version

Comment (ArangoGutierrez) [reply README.md:42]: this is part of `hack/prepare-release` ;P

**Review by marquiz — APPROVED** [review]
Sorry @ArangoGutierrez for the nitting. I didn't realize it was the release script doing the edits I complained about 🙈 Anyway, let's get this merged
/lgtm

**Pull request merged**